src/main/java/magentoPage/ProductPage.java

```diff
@@ -0,0 +1,108 @@
+    private By editCartButton = By.xpath("//a[@class='action viewcart']");
+    private final Properties testData = PropertyUtil.getProperties("testsData.properties");
+
+    private void openMenu() {
+        new Actions(driver)
+                .moveToElement(driver.findElement(showCartAction))
+                .pause(Duration.ofSeconds(1))
+                .click()
+                .perform();
+    }
+
+    public CartPage openCartPage() {
+        openMenu();
+        new WebDriverWait(driver, Duration.ofSeconds(Integer.parseInt(testData.getProperty("explicit.time"))))
+                .until(ExpectedConditions.elementToBeClickable(editCartButton));
+        driver.findElement(editCartButton).click();
+        return new CartPage(driver);
+    }
+
+    public ProductPage(WebDriver driver) {
+        this.driver = driver;
+        PageFactory.initElements(driver, this);
+    }
+
+    public ProductsPage backProductsPage() {
+        driver.navigate().back();
+        return new ProductsPage(driver);
+    }
+
+    public void addedProductToCart() {
+        new WebDriverWait(driver, Duration.ofSeconds(Integer.parseInt(testData.getProperty("explicit.time"))))
+                .until(ExpectedConditions.elementToBeClickable(addToCartButton));
+        addToCartButton.click();
+    }
+
+    public ProductCart getProduct() {
+
+
+        new WebDriverWait(driver, Duration.ofSeconds(Integer.parseInt(testData.getProperty("explicit.time"))))
+                .until(ExpectedConditions.textToBePresentInElement(productPrice, "$"));
+
+        ProductCart product = new ProductCart();
+        product.setSize(choiceRandomSize());
+        product.setColor(choiceRandomColor());
+        product.setName(productName.getText());
+        product.setPrice(productPrice.getText());
+        return product;
+    }
+
+    private String choiceRandomColor() {
+        List<WebElement> listColors = getColorOptions();
+        int sizeList = listColors.size();
+        int random = (int) (Math.random() * sizeList);
+        WebElement color = listColors.get(random);
+        color.click();
+        return color.getAttribute("option-label");
+    }
+
+    private String choiceRandomSize() {
+        List<WebElement> listSizes = getSizeOption();
+        int sizeList = listSizes.size();
+        int random = (int) (Math.random() * sizeList);
+        WebElement size = listSizes.get(random);
+        size.click();
+        return size.getText();
+    }
+
+    private List<WebElement> getSizeOption() {
+        new WebDriverWait(driver, Duration.ofSeconds(Integer.parseInt(testData.getProperty("explicit.time"))))
+                .until(ExpectedConditions.visibilityOf(sizeOptions));
+        return sizeOptions.findElements(By.xpath(".//div[contains(@class,'option')]"));
+    }
+
+    private List<WebElement> getColorOptions() {
+        return colorOptions.findElements(By.xpath(".//div[contains(@class,'option')]"));
```

Comment:
same xpath value used in two methods. Can be moved to class variables

Reply:
fixed